core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java

```diff
@@ -96,7 +94,7 @@ public void authenticationSucceedsWhenFirstProviderReturnsNullButSecondAuthentic
 
 	@Test(expected = IllegalArgumentException.class)
 	public void testStartupFailsIfProvidersNotSet() {
-		new ProviderManager(null);
```

Comment:
I think we should still have a test for `new ProviderManager((AuthenticationProvider) null)` and for `new ProviderManager((List<AuthenticationProvider>) null)`.

The reason is that an application might do:

```java
AuthenticationProvider provider = constructProvider();
return new ProviderManager(provider);
```

but `constructProvider` returns `null`.

I believe this also implies that the extra validation logic in `checkState()` will turn out to be necessary.

Reply:
Got it, thanks for the clarification. I have added the extra validation login in `checkState()` and tests for both null scenarios.